ivy_tests/test_ivy/test_functional/test_core/test_linalg.py

```diff
@@ -643,27 +643,39 @@ def test_matmul(*, x, y, test_flags, backend_fw, fn_name, on_device):
 
 # matrix_norm
 @handle_test(
+    ground_truth_backend="torch",
     fn_tree="functional.ivy.matrix_norm",
-    # ground_truth_backend="numpy",
     dtype_value_axis=helpers.dtype_values_axis(
         available_dtypes=helpers.get_dtypes("float"),
         min_num_dims=2,
         valid_axis=True,
         min_axes_size=2,
         max_axes_size=2,
+        max_value=1e4,
+        min_value=-1e4,
+        abs_smallest_val=1e-4,
         force_tuple_axis=True,
         allow_neg_axes=False,
+        safety_factor_scale="log",
+        large_abs_safety_factor=1.5,
+        small_abs_safety_factor=1.5,
     ),
     kd=st.booleans(),
     ord=st.sampled_from((-2, -1, 1, 2, -float("inf"), float("inf"), "fro", "nuc")),
+    dtypes=helpers.get_dtypes("float", none=True, full=False),
 )
 def test_matrix_norm(
-    *, dtype_value_axis, kd, ord, test_flags, backend_fw, fn_name, on_device
+    *, dtype_value_axis, kd, ord, dtypes, test_flags, backend_fw, fn_name, on_device
 ):
-    dtype, x, axis = dtype_value_axis
+    input_dtype, x, axis = dtype_value_axis
+    if dtypes[0] is not None:
+        dtypes[0] = input_dtype[0][0:-2] + max([input_dtype[0][-2:], dtypes[0][-2:]])
+
+    test_flags.test_gradients = True
```

Comment:
let's not hardcode `test_gradients` to `True` for each test and instead pass the one that was generated by the strategy